web/src/saveMiddleware.js

```diff
@@ -0,0 +1,26 @@
+import {
+  ADD_APD_ITEM,
+  ADD_APD_YEAR,
+  EDIT_APD,
+  REMOVE_APD_ITEM,
+  REMOVE_APD_YEAR
+} from './actions/editApd';
+import { saveApd } from './actions/apd';
+
+const saveMiddleware = (store, { save = saveApd } = {}) => next => action => {
```

Comment:
This makes my head hurt.

Reply:
Yeah... The way this redux middleware works is funky.